cmd/mdatagen/internal/templates/telemetrytest.go.tmpl

```diff
@@ -42,15 +42,45 @@ func (tt *Telemetry) AssertMetrics(t *testing.T, expected []metricdata.Metrics,
 	require.NoError(t, tt.Reader.Collect(context.Background(), &md))
 	// ensure all required metrics are present
 	for _, want := range expected {
-		got := getMetric(want.Name, md)
+		got := getMetricFromResource(want.Name, md)
 		metricdatatest.AssertEqual(t, want, got, opts...)
 	}
 
 	// ensure no additional metrics are emitted
 	require.Equal(t, len(expected), lenMetrics(md))
 }
 
-func getMetric(name string, got metricdata.ResourceMetrics) metricdata.Metrics {
+{{ range $name, $metric := .Telemetry.Metrics }}
+{{ if not $metric.Optional }}
```

Comment:
Should we generate for this as well?

FYI: I've never seen a usage of this optional. Can we check if used, otherwise to remove it?